### ISSUE TITLE
SDDM 0.16.0 not compatible with ConsoleKit1.x ?

### ISSUE THREAD

**Issue by purejoy** [sddm/sddm#903]
On a Gentoo box,  system got hung on the login screen after updating from 0.15 with ConsoleKit 1.2.
Found an error in the sddm.log:
```
[14:03:02.339] (II) DAEMON: Starting...                                                  
34995 [14:15:03.593] (WW) DAEMON: Signal received: SIGTERM                                     
34996 [14:15:03.676] (II) DAEMON: Initializing...                                              
34997 [14:15:03.680] (II) DAEMON: Console kit interface found                                  
34998 [14:15:03.680] (WW) DAEMON: **QDBusConnection: error: could not send message to service "or      g.freedesktop.ConsoleKit" path "/org/freedesktop/ConsoleKit/Manager" interface "/org.free      desktop.ConsoleKit.Manager" member "ListSeats": Invalid interface class: /org.freedesktop      .ConsoleKit.Manager**
```

**Comment by davidedmundson** [comment]
There is a code path to support legacy systems.
Frankly, it's untested.

From your output, it's found there is ConsoleKit, and then assuming we have a Logind style interface we can use.

That means we skip the very important

        if (DaemonApp::instance()->testing() || !Logind::isAvailable()) {
            //if we don't have logind/CK2, just create a single seat immediately and don't do any other connections
            createSeat(QStringLiteral("seat0"));
            return;
        }

It seems legacy CK uses the same service name, which is all I check.

**Comment by ekigwana** [comment]
Unfortunately, some of us cannot use systemd. @purejoy Gentoo provides sys-auth/elogind which is extracted from systemd if that is an option for you. Once you install that and add elogind to boost services you should then be ok.

**Comment by ruegerjr** [comment]
It appears that the code checks for logind, and if not found, executes an operatiion that requires features of logind!  It may not be practical with the current code base, but it seems to me that if there is no logind, no seat creation should occur.  At the very least continue to have a compile-time option to allow sddm to build and function for those without logind?

**Comment by purejoy** [comment]
@ekigwana thx! After migrating to elogind, everything runs smoothly.
I can also confirm that the error occurred with ConsoleKit2.

**Comment by Hoshpak** [comment]
I just tried sddm 0.16 with ConsoleKit2 on Void Linux and got the same error, sddm fails to start. I tried to call the dbus function manually:

```
# dbus-send --system --print-reply --dest="org.freedesktop.ConsoleKit" /org/freedesktop/ConsoleKit/Manager org.freedesktop.ConsoleKit.Manager.ListSeats
Error org.freedesktop.DBus.Error.AccessDenied: Rejected send message, 2 matched rules; type="method_call", sender=":1.60" (uid=1000 pid=32710 comm="dbus-send --system --print-reply --dest=org.freede") interface="org.freedesktop.ConsoleKit.Manager" member="ListSeats" error name="(unset)" requested_reply="0" destination="org.freedesktop.ConsoleKit" (uid=0 pid=1215 comm="console-kit-daemon --no-daemon ")
```

It assume that `ListSeats` comes from logind since I can't find any mention of it in the [ConsoleKit2 documentation](https://consolekit2.github.io/ConsoleKit2/#Manager). In ConsoleKit(2), there's function called `GetSeats` which seems to do the same thing for ConsoleKit(2):

```# dbus-send --system --print-reply --dest="org.freedesktop.ConsoleKit" /org/freedesktop/ConsoleKit/Manager org.freedesktop.ConsoleKit.Manager.GetSeats
method return time=1507754194.751161 sender=:1.11 -> destination=:1.61 serial=317 reply_serial=2
   array [
      object path "/org/freedesktop/ConsoleKit/Seat1"
   ]```

So the code needs to distinguish between logind and ConsoleKit(2) and call the proper methods. The same should go for `SeatNew` which is called `SeatAdded` in ConsoleKit.

**Comment by arklee** [comment]
Same problem,Now using startx instead

**Comment by kensington** [comment]
> There is a code path to support legacy systems.
> Frankly, it's untested.
> 
> From your output, it's found there is ConsoleKit, and then assuming we have a Logind style interface we can use.
> 
> That means we skip the very important
> 
>     if (DaemonApp::instance()->testing() || !Logind::isAvailable()) {
>         //if we don't have logind/CK2, just create a single seat immediately and don't do any other connections
>         createSeat(QStringLiteral("seat0"));
>         return;
>     }
> 
> It seems legacy CK uses the same service name, which is all I check.

It looks like src/daemon/LogindDBusTypes.cpp explicitly causes Logind::isAvailable to be true if CK is present, is that expected behaviour?

**Comment by davidedmundson** [comment]
It was intended; because ConsoleKit2 has a supposedly identical interface mapping.

However:
1) it evidently doesn't
2) this check doesn't distinguish CK1 & 2

**Comment by kensington** [comment]
In case it's useful for any other downstream, this patch has reportedly worked around the issue:
```--- a/src/daemon/LogindDBusTypes.cpp
+++ b/src/daemon/LogindDBusTypes.cpp
@@ -60,7 +60,7 @@
 
     if (QDBusConnection::systemBus().interface()->isServiceRegistered(QStringLiteral("org.freedesktop.ConsoleKit"))) {
         qDebug() << "Console kit interface found";
-        available = true;
+        available = false;
         serviceName = QStringLiteral("org.freedesktop.ConsoleKit");
         managerPath = QStringLiteral("/org/freedesktop/ConsoleKit/Manager");
         managerIfaceName = QStringLiteral("/org.freedesktop.ConsoleKit.Manager"); //note this doesn't match logind```

I haven't personally tested it though as I'm not using CK, and of course this is only a hack until a hopeful future fix in sddm/CK.

**Comment by davidedmundson** [comment]
Completely defeats the idea that CK2 does actually work properly as a logind interface and the whole point of the abstraction I put in...you may as well just drop the CK2 support at that point...but it'll definitely work.

Tell them to submit it.

**Comment by kensington** [comment]
Right, that's why I qualified it as a workaround and temporary hack. :-)

**Comment by ekigwana** [comment]
I opened an issue but have not heard back yet from CK2 devs.
